dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -345,7 +348,16 @@ def _update_categories(
         :type df_series: pandas.DataFrame
         :return: None
         """
-        category_count = df_series.value_counts(dropna=False).to_dict()
+        if self._cms_confidence:
+            category_count = defaultdict(int)
+            num_hashes = count_min_sketch.suggest_num_hashes(self._cms_confidence)
+            num_buckets = count_min_sketch.suggest_num_buckets(self._cms_relative_error)
+            cm = count_min_sketch(num_hashes, num_buckets)
+            for i in df_series:
```

Comment:
I'm concerned about looping through the entire if df_series is completely unique. That's where I think the top_k which utilizes a look-up of values that are repetitive.

Reply:
hmm... top_k is the obtained by sorting the category_count dict (obtained via cms, which requires a single pass over the series?)... 

Are you suggesting we still run the stopping condition for CMS?

Reply:
Sorry, I wasn't clear. I'm concerned about storage of the entire df_series in  `category_count`